tests/functional/h/views/admin/test_permissions.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+
+class TestAdminPermissions:
+    PAGES = (
+        # URL, accessible by role.Staff?
+        ("/admin/", True),
+        ("/admin/admins", False),
+        ("/admin/badge", False),
+        ("/admin/features", False),
+        ("/admin/groups", True),
+        ("/admin/mailer", True),
+        ("/admin/nipsa", False),
+        ("/admin/oauthclients", False),
+        ("/admin/organizations", True),
+        ("/admin/staff", False),
+        ("/admin/users", True),
+        ("/admin/search", False),
+    )
+
+    @pytest.mark.parametrize("url", (page[0] for page in PAGES))
+    def test_not_accessible_by_regular_user(self, app, url, user):
+        self.login(app, user)
+
+        res = app.get(url, expect_errors=True)
+
+        assert res.status_code == 404
+
+    @pytest.mark.parametrize("url", (page[0] for page in PAGES))
+    def test_accessible_by_admin(self, app, url, admin_user):
+        self.login(app, admin_user)
+
+        res = app.get(url)
+
+        assert res.status_code == 200
```

Comment:
I think just this is enough:

```suggestion
        app.get(url)
```

`get()` will fail by default if the response isn't 200 or 3xx. If you want to assert that it's 200 and not a redirect you can do `app.get(url, status=200)`.

Reply:
I don't think it matters in this case actually. We only care it's not a 404.